module/documents/item.mjs

```diff
@@ -470,6 +470,48 @@ export default class Item5e extends Item {
 
   /* -------------------------------------------- */
 
+  /**
+   * Determine an item's proficiency level based on its parent actor's proficiencies.
+   * @protected
+   */
+  _prepareProficiency() {
+    if ( !["spell", "weapon", "equipment", "tool"].includes(this.type) ) return;
+    if ( !this.actor?.system.attributes?.prof ) {
+      this.system.prof = new Proficiency(0, 0);
+      return;
+    }
+
+    let multiplier;
+    const tools = this.actor.system.tools;
+    const traits = this.actor.system.traits;
+
+    // Always considered proficient in spell attacks.
+    if ( this.type === "spell" ) multiplier = 1;
+
+    // Proficiency overridden at the item level.
+    else if ( this.system.proficient !== null ) multiplier = this.system.proficient;
+
+    // Otherwise, determine proficiency based on the actor's proficiencies.
+    else {
+      if ( this.type === "tool" ) {
+        const baseItemProf = tools?.[this.system.baseItem];
+        const categoryProf = tools?.[this.system.toolType];
+        multiplier = Math.max(baseItemProf?.value ?? 0, categoryProf?.value ?? 0);
+      } else {
+        const type = this.type === "equipment" ? this.system.armor?.type : this.system.weaponType;
+        const config = CONFIG.DND5E[`${this.type === "equipment" ? "armor" : "weapon"}ProficienciesMap`];
+        const itemProf = config[type];
+        const actorProfs = traits?.[this.type === "equipment" ? "armorProf" : "weaponProf"]?.value ?? new Set();
+        const isProficient = (itemProf === true) || actorProfs.has(itemProf) || actorProfs.has(this.system.baseItem);
+        multiplier = Number(isProficient);
+      }
+    }
```

Comment:
All this can be brought up a level.

```suggestion
    // Tools based on actor's proficiencies.
    else if ( this.type === "tool" ) {
      const baseItemProf = tools?.[this.system.baseItem];
      const categoryProf = tools?.[this.system.toolType];
      multiplier = Math.max(baseItemProf?.value ?? 0, categoryProf?.value ?? 0);
    }

    // Otherwise, determine proficiency based on the actor's proficiencies.
    else {
      const type = this.type === "equipment" ? this.system.armor?.type : this.system.weaponType;
      const config = CONFIG.DND5E[`${this.type === "equipment" ? "armor" : "weapon"}ProficienciesMap`];
      const itemProf = config[type];
      const actorProfs = traits?.[this.type === "equipment" ? "armorProf" : "weaponProf"]?.value ?? new Set();
      const isProficient = (itemProf === true) || actorProfs.has(itemProf) || actorProfs.has(this.system.baseItem);
      multiplier = Number(isProficient);
    }
```